InMobi/src/main/java/com/mopub/mobileads/InMobiAdapterConfiguration.kt

```diff
@@ -0,0 +1,243 @@
+                    } else {
+                        consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, false)
+                    }
+                } ?: run {
+                    consentObject.put(CONSENT_KEY, "0")
+                    consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, false)
+                }
+            } catch (e: JSONException) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM_WITH_THROWABLE, "InMobi cannot obtain consent because of error: ", e)
+            }
+
+            return consentObject
+        }
+
+        fun initialiseInMobi(configuration: Map<String, String>, context: Context, initCompletionListener: InitCompletionListener) {
+            try {
+                val accountId = getAccountId(configuration)
+                val consentObject = InMobiGDPR.gdprConsentDictionary ?: run {
+                    getInMobiConsentDictionary()
+                }
+                InMobiSdk.init(context, accountId, consentObject) {
+                    if (it == null) {
+                        isInMobiSdkInitialised = true
+                        initCompletionListener.onSuccess()
+                    } else {
+                        initCompletionListener.onFailure(it, null)
+                    }
+                }
+            } catch (e: Exception) {
+                initCompletionListener.onFailure(null, e)
+            }
+        }
+
+        // TODO: Revision needed
+        fun getMoPubErrorCode(statusCode: InMobiAdRequestStatus.StatusCode?): MoPubErrorCode {
+            return when (statusCode) {
+                InMobiAdRequestStatus.StatusCode.INTERNAL_ERROR -> MoPubErrorCode.INTERNAL_ERROR
+                InMobiAdRequestStatus.StatusCode.REQUEST_INVALID -> MoPubErrorCode.NETWORK_INVALID_STATE
+                InMobiAdRequestStatus.StatusCode.NETWORK_UNREACHABLE -> MoPubErrorCode.NO_CONNECTION
+                InMobiAdRequestStatus.StatusCode.NO_FILL -> MoPubErrorCode.NO_FILL
+                InMobiAdRequestStatus.StatusCode.REQUEST_TIMED_OUT -> MoPubErrorCode.NETWORK_TIMEOUT
+                InMobiAdRequestStatus.StatusCode.SERVER_ERROR -> MoPubErrorCode.NETWORK_INVALID_STATE
+                else -> MoPubErrorCode.UNSPECIFIED
+            }
+        }
+
+        fun getAccountId(dict: Map<String, String>): String {
+            val accountIdErrorMessage = "Please make sure you provide correct Account ID information on MoPub UI."
+            val accountIdString: String? = dict[ACCOUNT_ID_KEY]
+            if (accountIdString.isNullOrEmpty()) {
+                throw Exception("InMobi Account ID parameter is null or empty. " +
+                        accountIdErrorMessage)
+            } else {
+                return accountIdString
+            }
+        }
+
+        fun getPlacementId(dict: Map<String, String>): Long {
+            val placementIdErrorMessage = "Please make sure you provide correct Placement ID information on MoPub UI."
```

Comment:
One more thing can be done, can be made static or constant at class level, because over here it will initialize again and again whenever this method is called. And frequency of this is going to be very high.